tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_cplx_recip.py

```diff
@@ -0,0 +1,48 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+import pytest
+import tt_lib
+from tests.tt_eager.python_api_testing.unit_testing.backward_ops.utility_funcs import compare_results
+
+
+@pytest.mark.parametrize(
+    "input_shapes",
+    (
+        (torch.Size([1, 1, 32, 32])),
+        (torch.Size([1, 1, 320, 384])),
+        (torch.Size([1, 3, 320, 384])),
+    ),
+)
+def test_bw_complex_recip_bw(input_shapes, device):
+    torch.manual_seed(0)
+    in_data = torch.randn(input_shapes, dtype=torch.complex64)
+    in_data.requires_grad = True
+
+    torch.manual_seed(42)
+    grad_data = torch.randn(input_shapes, dtype=torch.complex64)
+
+    in_data_cplx = torch.cat((in_data.real, in_data.imag), dim=3)
+    input_tensor = (
+        tt_lib.tensor.Tensor(in_data_cplx, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
+    )
+    grad_data_cplx = torch.cat((grad_data.real, grad_data.imag), dim=3)
+    grad_tensor = (
+        tt_lib.tensor.Tensor(grad_data_cplx, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
+    )
+    tt_output_tensor_on_device = tt_lib.tensor.complex_recip_bw(grad_tensor, input_tensor)
```

Comment:
I'm not following why we are not able to make this op part of tt_lib.tensor.reciprocal_bw ?
What am I missing here?

Reply:
I do see that we have a tt_lib.tensor.complex_recip and this is the backward direction for this op.    Why does torch have this implemented with reciprocal but we don't?

Reply:
@eyonland the complex ops were already implemented separately ( I am not sure why) and complex tensors have a shape constraint on the last dim as they include both real and imaginary part.

Reply:
I'm approving given the intent is to simply provide the backward.   Raising this question internally as I'm not clear myself why we have this.

Reply:
@eyonland In TT we have two methods of complex implementation names type1 and type2
Type1 - Given tensor is split into two parts real and imaginary assuming both are in the last dimension. Example if tensor shape is 1,1, 32, 64 --> 1,1,32,[0-31] is real part and 1,1,32, [32-63] imaginary part

Type2 - we have a class called ComplexTensor (array of tensors) 

We are not sure which one will be the final version and also we are not sure how to differentiate between normal tensor and complex tensor at the initial development phase. Hence it is developed as separate op